### ISSUE TITLE
Add workspace `vscode` configuration to enable consistent `formatOnSave` and linting.

### ISSUE THREAD

**Issue by MattyMc** [Virtual-Coffee/virtualcoffee.io#411]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Issue Context

PR #401 adds official prettier settings into the VC repo. Prettier, along with the js libraries `lint-staged` and `husky` have  surfaced a number of benefits to code quality and consistency. The three primary interfaces to these new capabilities are: 

1. Manually running prettier from the CLI to scan the entire repo: `yarn format`;
2. Manually running prettier from the CLI to scan staged files: `yarn lint-staged`;
3. Automatically running prettier on commit.

All the commands above will attempt to fix any syntax errors automatically and will surface warnings to the terminal if the error cannot be addressed automatically. 

However, although the addition of prettier has many benefits, the implementation leaves a few opportunities and drawbacks: 

1. To see the changes prettier will make to files you have to run a manual CLI command (1 or 2 above). If a dev relies on prettier to automatically run, the style changes are made automatically and then immediately committed. The committed files would then need to be manually checked against a previous commit to see the changes prettier made.
2. Some developers may have particular settings in the prettier vscode extension, `prettier-vscode`, that override prettier's workspace settings (e.x. a custom parser for particular file extensions). If formatOnSave is turned on, the vscode extension would autoformat with the user's rules, and on commit format with the project's rules resulting in potentially messy git diffs locally. 
3. Autoformatting on save and linting are excellent learning opportunity for new developers to improve the syntax and style of the code they write. For example, syntax errors such as unclosed tags would be immediately visible instead of having to wait until commit or manually running the CLI commands.
4. Development speed would increase as a result of being able to see syntax errors immediately on save.

### Proposed solution

The proposed solution is to create a minimal `settings.json` and `extensions.json` that enable vscode users to have syntax errors highlighted in the files they're working on (eslint) and have files autoformat on save (prettier).

VSCode supports both [user settings and workspace settings](https://code.visualstudio.com/docs/getstarted/settings) for this purpose. From the [documentation](https://code.visualstudio.com/docs/getstarted/settings):

- User Settings: Settings that apply globally to any instance of VS Code you open.
- Workspace Settings: Settings stored inside your workspace and only apply when the workspace is opened.

Thus, workspace settings can be added without affecting any of the developer's settings outside of VSCode, and developers who use a separate editor will not be affected.

Proposed steps: 

- [ ] Add recommended extensions for `eslint` and `prettier` so that new users are prompted to install when they open the VC repo. 
- [ ] Write a base, minimal configuration for eslint such that it follows prettier's rules (`.estlint` in base dir).
- [ ] Add a minimal rule set for prettier's vscode extension to ensure all files (include `njk`) are parsed consistently on save.

In short, the above improvements will only serve to make more transparent and convenient the changes that are already happening with prettier through the CLI or on commit. 

### Code of Conduct

- [X] I've read the Code of Conduct and understand my responsibilities as a member of the Virtual Coffee community

**Comment by MattyMc** [comment]
@danieltott and @nickytonline would love if you'd weigh in on this. I'd love to take this one on, too. I recently went through this process with my team and would like to contribute this work to open source (potentially write a little blog post, too).

**Comment by MattyMc** [comment]
Hey @danieltott: The [VSCode repo](https://github.com/microsoft/vscode/tree/main/.vscode) is something we used for inspiration. I don't think there are a ton of code repos using this right now, so it may make sense not to prioritize this work until it's more common. It is something I'd expect to increase. With more repos adding standardized code formatting as a pre-commit step, it only makes sense to bring that auto-formatting into the dev process, too.

**Comment by MattyMc** [comment]
@danieltott let me mock something up. We've done a ton of work on this already, and at the very least we can try to add some stuff in and see if it helps folks.

**Comment by nickytonline** [comment]
We use recommended extensions, and debug configurations at Forem, but we don't enforce stuff like formatOnSave because that is a user preference. What we do though for lining and formatting is use husky/lint-staged as a precommit hook.

- https://github.com/forem/forem/blob/main/.vscode/extensions.json
- https://github.com/forem/forem/blob/main/.vscode/launch.json
